src/SharedMemoryParcel.php

```diff
@@ -0,0 +1,481 @@
+ * should be protected with a mutex to preserve data integrity.
+ *
+ * When used with forking, the object must be created prior to forking for both
+ * processes to access the synchronized object.
+ *
+ * @see http://php.net/manual/en/book.shmop.php The shared memory extension.
+ * @see http://man7.org/linux/man-pages/man2/shmctl.2.html How shared memory works on Linux.
+ * @see https://msdn.microsoft.com/en-us/library/ms810613.aspx How shared memory works on Windows.
+ *
+ * @template T
+ * @template-implements Parcel<T>
+ */
+final class SharedMemoryParcel implements Parcel
+{
+    /** @var int The byte offset to the start of the object data in memory. */
+    private const MEM_DATA_OFFSET = 7;
+
+    private const MAX_ID = 0xffffffff;
+
+    // A list of valid states the object can be in.
+    private const STATE_UNALLOCATED = 0;
+    private const STATE_ALLOCATED = 1;
+    private const STATE_MOVED = 2;
+    private const STATE_FREED = 3;
+
+    private static int $nextId = 0;
+
+    /**
+     * @param string $id
+     * @param mixed $value
+     * @param int $size The initial size in bytes of the shared memory segment. It will automatically be
+     *     expanded as necessary.
+     * @param int $permissions Permissions to access the semaphore. Use file permission format specified as
+     *     0xxx.
+     * @param Serializer|null $serializer
+     *
+     * @return self
+     *
+     * @throws ParcelException
+     * @throws SyncException
+     * @throws \Error If the size or permissions are invalid.
+     */
+    public static function create(
+        Mutex $mutex,
+        mixed $value,
+        int $size = 8192,
+        int $permissions = 0600,
+        ?Serializer $serializer = null
+    ): self {
+        if ($size <= 0 || $size > 1 << 27) {
+            throw new \Error('The memory size must be greater than 0 and less than 128 MB');
+        }
+
+        if ($permissions <= 0 || $permissions > 0777) {
+            throw new \Error('Invalid permissions');
+        }
+
+        $parcel = new self(0, $mutex, $serializer);
+        $parcel->init($value, $size, $permissions);
+        return $parcel;
+    }
+
+    /**
+     * @param Mutex $mutex
+     * @param int $key
+     * @param Serializer|null $serializer
+     *
+     * @return self
+     *
+     * @throws ParcelException
+     */
+    public static function use(Mutex $mutex, int $key, ?Serializer $serializer = null): self
+    {
+        $parcel = new self($key, $mutex, $serializer);
+        $parcel->open();
+        return $parcel;
+    }
+
+    private static function makeKey(string $id): int
+    {
+        return (int) \abs(\unpack("l", \md5($id, true))[1]);
+    }
+
+    /** @var int The shared memory segment key. */
+    private int $key;
+
+    /** @var Mutex A mutex for synchronizing on the parcel. */
+    private Mutex $mutex;
+
+    /** @var \Shmop An open handle to the shared memory segment. */
+    private ?\Shmop $handle = null;
```

Comment:
On which version has it been converted to an object?

Reply:
8.0